### DISCUSSION TITLE
Improve local DNS with Spin

### DISCUSSION THREAD

**Discussion by jaydrogers** [serversideup/spin#102] [Ideas]
## Problem
- The local DNS configuration in the docs could use some extra clarity

## Related threads
https://discord.com/channels/910287105714954251/910299290230997003/1195043928790007878

## Proposed Solution
- [ ] Identify which areas could use improvement
- [ ] Ensure the talk about Traefik labels are discussed
- [ ] See if there is a way we can automated (hopeful feature 🦄)

**Comment by yusufkandemir** [comment]
I've found [`dnsmasq`](https://thekelleys.org.uk/dnsmasq/doc.html) to work well.

macOS instructions for installing it and configuring it for `*.test` domains
```bash
# install dnsmasq
brew install dnsmasq

# create the configuration and point *.test to 127.0.0.1
mkdir -pv $(brew --prefix)/etc/
echo 'address=/.test/127.0.0.1' >> $(brew --prefix)/etc/dnsmasq.conf

# create resolver for *.test domains
sudo mkdir -p /etc/resolver
sudo bash -c 'echo "nameserver 127.0.0.1" > /etc/resolver/test

# start the dnsmasq service, will also be added to system startup
sudo brew services start dnsmasq
```
